### PR TITLE
Fix negotiate auto invalid skips

### PR DESCRIPTION
**Pull request by robert-w-gries**
### Test Results

`test_duplex`, `test_negotiate_auto_ethernet`, and `test_negotiate_auto_portchannel` pass for the following platforms:
1. n3k
2. n5k
3. n6k
4. n7k
5. n8k
   - physical
   - virtual
6. n9k
   - camden - physical
   - camden - virtual
   - dublin - physical
### Portchannel Error Fix

The [unnecessary XR `UnsupportedError` check](https://github.com/cisco/cisco-network-node-utils/compare/develop...robert-w-gries:feature/negotiate-auto-fix#diff-cf2422f8b2cb8aa033a2f17f39d7354fL768) was causing an issue when re-running the test:

``` bash
  1) Error:
TestInterface#test_negotiate_auto_portchannel:
Cisco::CliError: [interface_channel_group ethernet1/1] The command ' channel-group 10 force' was rejected with error:
command failed: port not compatible [speed]
```

### CODE REVIEW

File: tests/test_interface.rb [hunk 1]
```diff
@@ -707,68 +705,37 @@ def test_svi_prop_nil_when_ethernet
...
-    # 10G+ interfaces do not support negotiation
-    interface_supports_property?(intf, e.message)
-  end
 
-  # Yes, this method is nearly identical to negotiate_true.
-  # The negotiate property is evil to troubleshoot. Keep them separate.
-  def negotiate_false(interface, speed)
-    # puts "false: 'speed #{speed}', 'no negotiate auto'"
-    intf = interface.name
+    # Note that 'speed' and 'negotiate' are tightly coupled
```
Comment (mikewiebe) [inline tests/test_interface.rb]: _nit_ `Note that 'speed' and 'negotiate auto' are tightly coupled`

File: tests/test_interface.rb [hunk 2]
```diff
@@ -707,68 +705,37 @@ def test_svi_prop_nil_when_ethernet
...
-           "#{intf} negotiate auto value should be false")
-    assert_show_match(pattern: /^\s+no negotiate auto/)
-  rescue Cisco::CliError => e
-    # 10G+ interfaces do not support negotiation
-    interface_supports_property?(intf, e.message)
+    if speed == 'auto'
+      interface.negotiate_auto = true
+      assert(interface.negotiate_auto,
+             "#{interface.name} negotiate auto value should be true")
+      assert_show_match(pattern: pattern)
```
Comment (mikewiebe) [inline tests/test_interface.rb]: Do you feel like the assert_show_match and refute_show_match below add significant value?  IMO, the assert and refute are sufficient.

File: tests/test_interface.rb [hunk 3]
```diff
@@ -707,68 +705,37 @@ def test_svi_prop_nil_when_ethernet
...
-      member.channel_group = 10
-    rescue Cisco::UnsupportedError
-      # Some XR platform/version combinations don't support port-channel intfs
-      skip('bundle id config not supported on this node') if platform == :ios_xr
-      raise
+    if validate_property_excluded?('interface_channel_group', 'channel_group')
+      member = InterfaceChannelGroup.new(interfaces[0])
+      assert_raises(Cisco::UnsupportedError) do
+        member.channel_group = 10
+      end
```
Comment (mikewiebe) [inline tests/test_interface.rb:735]: Should we return from the test here?

Comment (robert-w-gries) [inline tests/test_interface.rb:735]: Yep.  Added a `return`

**Comment by robert-w-gries** [comment]
Comments addressed

**Comment by mikewiebe** [comment]
👍

**Pull request merged**